File: fatorial.py
def fatorial(num):
    if num == 1:
        return 1
    else:
        return num * fatorial(num - 1)

if __name__ == '__main__':
    a = input("Digite um número: ")
    print(f'O fatorial de {a} é {fatorial(int(a))}')

File: fatorial.c
#include<stdio.h>
#include<stdlib.h>

unsigned long long int fatorial(unsigned int num){
    if (num == 1 || num == 0) {
        return 1;
    } else {
        return num * fatorial(num - 1);
    }
    
}

int main(void){
    unsigned int a;
    printf("Digite um número: ");
    scanf("%u", &a);
    printf("O fatorial de %u é %llu\n", a, fatorial(a));
    return EXIT_SUCCESS;
}